BDArmory/Control/MissileFire.cs

```diff
@@ -2967,14 +3136,16 @@ IEnumerator GuardBombRoutine()
                         }
                     }
                 }
-
                 if (targetDistSqr > radiusSqr
                     || Vector3.Dot(vessel.up, vessel.transform.forward) > 0) // roll check
                 {
-                    if (targetDistSqr > Mathf.Max(4f * radiusSqr, 40000f) && // Check for overshooting the target by more than twice the blast radius or 200m
-                        Vector3.Dot(guardTarget.CoM - bombAimerCPA, guardTarget.CoM - transform.position) < 0)
+                    Vector3 upDirection = VectorUtils.GetUpDirection(vessel.CoM);
+                    if ((leadTarget - vessel.CoM).sqrMagnitude < (pilotAI.extendDistanceAirToGround * pilotAI.extendDistanceAirToGround) && //Check the target is within bombing run dist,
+                        ((targetDistSqr > Mathf.Max(4f * radiusSqr, pilotAI.divebombing ? 1000000 : 40000f) && Vector3.Dot((leadTarget - bombAimerPosition).ProjectOnPlanePreNormalized(upDirection), (leadTarget - vessel.CoM).ProjectOnPlanePreNormalized(upDirection)) < 0) // not overshooting the target by more than twice the blast radius or 200m if levelbombing, 1km if divebombing,
+                        || (targetDistSqr < 4 * radiusSqr && Vector3.Dot(vessel.up, vessel.transform.forward) > 0))) //or on final approach and upside down
                     {
-                        pilotAI.RequestExtend("too close to bomb", guardTarget, minDistance: (vessel.CoM - guardTarget.CoM).magnitude + (pilotAI ? pilotAI.extendDistanceAirToGround : 2000f), ignoreCooldown: true); // Extend from target vessel by an extra 2km for a reasonable bombing run.
+                        if (pilotAI.extendingReason != "too close to bomb") //don't spam this every frame
+                             pilotAI.RequestExtend("too close to bomb", guardTarget, minDistance: pilotAI.extendDistanceAirToGround + ((float)vessel.horizontalSrfSpeed * (BDAMath.Sqrt(2 * (float)vessel.altitude / (float)FlightGlobals.getGeeForceAtPosition(vessel.CoM).magnitude))), ignoreCooldown: true); // Extend from target vessel by A2G extendDist + distance bomb would cover while falling
```

Comment:
``pilotAI.RequestExtend("too close to bomb", guardTarget, minDistance: (vessel.CoM - guardTarget.CoM).magnitude + (pilotAI ? pilotAI.extendDistanceAirToGround : 2000f), ignoreCooldown: true);`` is only ever going to use the pilotAI.extendDistanceAirToGround, the pilotAI ? check is always going to return true, since it's inside pilotAI.RequestExtend().
VTOL AI doesn't currently have support for external extend commands

That said, the additional distance the bomb is (roughly) estimated to fall is also being added to the extendDistance in the pilotAI CheckExtend, so having it here is a doubling of that fall dist.

Reply:
> is only ever going to use the pilotAI.extendDistanceAirToGround, the pilotAI ? check is always going to return true, since it's inside pilotAI.RequestExtend().

Yep, I missed that. However, the point is that generally (though not always, so that logic should probably be looked at too - I suspect there's a race condition between the AI and WM), the AI starts with extending due to "Surface target" in `CheckExtend` and then if that isn't enough the "too close to bomb" gets triggered here and the plane needs to extend a bit further than it's current distance (which should be the A2G + rough additional fall distance), so setting the requested extend `minDistance` to that fixed distance could end up with it getting stuck in an extending loop if the rough estimate is particularly bad. We could have the additional distance be chosen a bit smarter than simply an extra A2G extend distance though.

I figured that we'd want to do something with the VTOL AI dropping bombs at some point since someone is bound to ask for it. I doubt that it'll ever be relevant for the surface or orbital AIs though.